### DISCUSSION TITLE
5G Non-terrestrial networks

### DISCUSSION THREAD

**Discussion by yamrodest** [NVlabs/sionna#193] [Q&A]
Hi!

I want to simulate a 5G NTN scenario. Do you think that Sionna has capabilities for that? What modifications should be considered for the PUSCHTransmitter and PUSCHReceiver?

On the other hand, from my first research, the NTN TDL channel defined in 3GPP TR 38.811 Section 6.9.2 could be configured using the Sionna's TDL implementation. Do you think that is possible or would a new implementation for the NTN channel be better?

Thank you in advance.

**Comment by jhoydis** [comment]
Hi,

Thanks for this interesting question. First of all, I barely know 5G NTN. 

Concerning the PUSCHTransmitter and Receiver, I am not sure how different they are for NTN.

After a quick look at the TR 38.811, it seems that the channel models are very similar (from an implementation perspective) to the 38.901 models. Concerning the TDL models, the biggest change seems to come from the additional Doppler shift due to satellite motion. It would be very easy to use a custom PDP that could be provided in JSON format ([example](https://github.com/NVlabs/sionna/blob/main/sionna/channel/tr38901/models/TDL-A.json)).
Overall, adding the TDL A-D models for NTN seems rather straight-forward. Definitely something we could add as feature in a future release. 

If you would like to give it a try yourself, you would only need to make modifications to the [TDL class](https://github.com/NVlabs/sionna/blob/main/sionna/channel/tr38901/tdl.py).
